packages/amazon-ecs-cni-plugins/amazon-ecs-cni-plugins.spec

```
@@ -0,0 +1,138 @@
+%{summary}.
+
+%package ecs-agent-extras
+Summary: Extra files necessary for the ECS agent
+Provides: %{name}(ecs-agent-extras)
+Requires: (%{_cross_os}image-feature(no-fips) and %{name}(binaries))
+Conflicts: (%{_cross_os}image-feature(fips) or %{name}-ecs-agent-fips-extras)
+
+%description ecs-agent-extras
+%{summary}.
+
+%package ecs-agent-fips-extras
+Summary: Extra files necessary for the ECS agent, FIPS edition
+Provides: %{name}(ecs-agent-extras)
+Requires: (%{_cross_os}image-feature(fips) and %{name}(binaries))
+Conflicts: (%{_cross_os}image-feature(no-fips) or %{name}-ecs-agent-extras)
+
+%description ecs-agent-fips-extras
+%{summary}.
+
+%prep
+%autosetup -n %{ecscni_gorepo}-%{ecscni_gitrev} -p1
+
+# Symlink amazon-ecs-cni-plugins-%{ecscni_gitrev} to the GOPATH location
+%cross_go_setup %{ecscni_gorepo}-%{ecscni_gitrev} %{ecscni_goproject} %{ecscni_goimport}
+
+%build
+# cross_go_configure cd's to the correct GOPATH location
+%cross_go_configure %{ecscni_goimport}
+LD_ECS_CNI_VERSION="-X github.com/aws/amazon-ecs-cni-plugins/pkg/version.Version=$(cat VERSION)"
+ECS_CNI_HASH="%{ecscni_gitrev}"
+LD_ECS_CNI_SHORT_HASH="-X github.com/aws/amazon-ecs-cni-plugins/pkg/version.GitShortHash=${ECS_CNI_HASH::8}"
+LD_ECS_CNI_PORCELAIN="-X github.com/aws/amazon-ecs-cni-plugins/pkg/version.GitPorcelain=0"
+
+declare -a ECS_CNI_BUILD_ARGS
+ECS_CNI_BUILD_ARGS=(
+  -ldflags "${GOLDFLAGS} ${LD_ECS_CNI_VERSION} ${LD_ECS_CNI_SHORT_HASH} ${LD_ECS_CNI_PORCELAIN}"
+)
+
+go build "${ECS_CNI_BUILD_ARGS[@]}" -o ecs-eni ./plugins/eni
+gofips build "${ECS_CNI_BUILD_ARGS[@]}" -o fips/ecs-eni ./plugins/eni
+
+go build "${ECS_CNI_BUILD_ARGS[@]}" -o ecs-ipam ./plugins/ipam
+gofips build "${ECS_CNI_BUILD_ARGS[@]}" -o fips/ecs-ipam ./plugins/ipam
+
+go build "${ECS_CNI_BUILD_ARGS[@]}" -o ecs-bridge ./plugins/ecs-bridge
+gofips build "${ECS_CNI_BUILD_ARGS[@]}" -o fips/ecs-bridge ./plugins/ecs-bridge
+
+%install
+install -d %{buildroot}%{_cross_libexecdir}
+install -D -p -m 0755 ecs-bridge %{buildroot}%{_cross_libexecdir}/cni/ecs/ecs-bridge
+install -D -p -m 0755 ecs-eni %{buildroot}%{_cross_libexecdir}/cni/ecs/ecs-eni
+install -D -p -m 0755 ecs-ipam %{buildroot}%{_cross_libexecdir}/cni/ecs/ecs-ipam
+
+install -d %{buildroot}%{_cross_fips_libexecdir}
+install -D -p -m 0755 fips/ecs-bridge %{buildroot}%{_cross_fips_libexecdir}/cni/ecs/ecs-bridge
+install -D -p -m 0755 fips/ecs-eni %{buildroot}%{_cross_fips_libexecdir}/cni/ecs/ecs-eni
+install -D -p -m 0755 fips/ecs-ipam %{buildroot}%{_cross_fips_libexecdir}/cni/ecs/ecs-ipam
+
+# Create symlinks to ECS CNI plugin binaries for amazon-ecs-agent
+install -d %{buildroot}{%{_cross_libexecdir},%{_cross_fips_libexecdir}}/amazon-ecs-agent
+for p in \
+  ecs-bridge \
+  ecs-eni \
+  ecs-ipam \
+; do
+  ln -rs %{buildroot}%{_cross_libexecdir}/cni/ecs/${p} %{buildroot}%{_cross_libexecdir}/amazon-ecs-agent/${p}
+  ln -rs %{buildroot}%{_cross_libexecdir}/cni/ecs/${p} %{buildroot}%{_cross_fips_libexecdir}/amazon-ecs-agent/${p}
```

We should create symlinks that stay inside `/usr/fips/libexec`, so that they work whether or not the overlay mount is in place at runtime.

```suggestion
  ln -rs %{buildroot}%{_cross_fips_libexecdir}/cni/ecs/${p} %{buildroot}%{_cross_fips_libexecdir}/amazon-ecs-agent/${p}
```

However, if that causes the symlinks to be broken for the overlay mount, then we should only create the ones for `/usr/libexec`, and rely on the FIPS overlay mount to make them point to the right binaries. In that case, we wouldn't have separate FIPS or non-FIPS versions, that just invites confusion.

---

Got it, let me test this

---

Suggested change works with the `aws-ecs-2` variant!